x-pack/apm-server/aggregation/txmetrics/aggregator_test.go

```diff
@@ -219,6 +219,71 @@ func TestAggregatorRunPublishErrors(t *testing.T) {
 	}
 }
 
+func TestAggregateRepresentativeCount(t *testing.T) {
+	reqs := make(chan publish.PendingReq, 1)
+
+	agg, err := txmetrics.NewAggregator(txmetrics.AggregatorConfig{
+		Report:                         makeChanReporter(reqs),
+		MaxTransactionGroups:           1,
+		MetricsInterval:                time.Microsecond,
+		HDRHistogramSignificantFigures: 1,
+		RUMUserAgentLRUSize:            1,
+	})
+	require.NoError(t, err)
+
+	// Record a transaction group so subsequent calls yield immediate metricsets,
+	// and to demonstrate that fractional transaction counts are accumulated.
+	agg.AggregateTransaction(&model.Transaction{Name: "fnord", RepresentativeCount: 1})
+	agg.AggregateTransaction(&model.Transaction{Name: "fnord", RepresentativeCount: 1.5})
+
+	for _, test := range []struct {
+		representativeCount float64
+		expectedCount       int64
+	}{{
+		representativeCount: 0,
+		expectedCount:       1,
+	}, {
+		representativeCount: -1,
+		expectedCount:       1,
+	}, {
+		representativeCount: 2,
+		expectedCount:       2,
+	}, {
+		representativeCount: 1.50, // round half away from zero
+		expectedCount:       2,
+	}} {
+		m := agg.AggregateTransaction(&model.Transaction{
+			Name:                "foo",
+			RepresentativeCount: test.representativeCount,
+		})
+		require.NotNil(t, m)
+
+		m.Timestamp = time.Time{}
+		assert.Equal(t, &model.Metricset{
+			Metadata:             model.Metadata{},
+			TimeseriesInstanceID: ":foo:1db641f187113b17",
+			Transaction: model.MetricsetTransaction{
+				Name: "foo",
+				Root: true,
+			},
+			Samples: []model.Sample{{
+				Name:   "transaction.duration.histogram",
+				Counts: []int64{test.expectedCount},
+				Values: []float64{0},
+			}},
+		}, m)
+	}
+
+	stopAggregator := runAggregator(agg)
+	defer stopAggregator()
+
+	req := expectPublish(t, reqs)
+	require.Len(t, req.Transformables, 1)
+	metricset := req.Transformables[0].(*model.Metricset)
+	require.Len(t, metricset.Samples, 1)
+	assert.Equal(t, []int64{3 /*round(1+1.5)*/}, metricset.Samples[0].Counts)
```

Comment:
Do you mind adding a comment explaining this check for `metricset`? I find it hard to follow at the end of this test.

Reply:
I've added a comment. If it's still unclear let me know, I'm happy to do a follow up.